controllers/cloudstackfailuredomain_controller_test.go

```diff
@@ -49,14 +49,120 @@ var _ = Describe("CloudStackFailureDomainReconciler", func() {
 					arg1.(*infrav1.CloudStackZoneSpec).Network.Type = cloud.NetworkTypeShared
 				}).MinTimes(1)
 
-			tempfd := &infrav1.CloudStackFailureDomain{}
+		})
+
+		It("Should set failure domain Status.Ready to true.", func() {
+			assertFailuredomainCreated()
+		})
+		It("Should delete failure domain if no VM under this failure domain.", func() {
+			assertFailuredomainCreated()
+			Ω(k8sClient.Delete(ctx, dummies.CSFailureDomain1))
+
+			assertFailuredomainNotExisted()
+		})
+		It("Should trigger machine deployment rollout when only worker node in failure domain.", func() {
+			assertFailuredomainCreated()
+			setupCAPIMachineDeploymentCRDs(dummies.CAPIMachineDeployment)
+			setupCAPIMachineAndCSMachineCRDs(dummies.CSMachine1, dummies.CAPIMachine1)
+			setMachineOwnerReference(dummies.CSMachine1, dummies.MachineSetOwnerRef)
+			labelMachineFailuredomain(dummies.CSMachine1, dummies.CSFailureDomain1)
+			labelMachineDeploymentName(dummies.CSMachine1, dummies.CAPIMachineDeployment.Name)
+			Ω(k8sClient.Delete(ctx, dummies.CSFailureDomain1))
+
+			tempMd := &clusterv1.MachineDeployment{}
+			Eventually(func() bool {
+				key := client.ObjectKeyFromObject(dummies.CAPIMachineDeployment)
+				if err := k8sClient.Get(ctx, key, tempMd); err == nil {
+					_, found := tempMd.Spec.Template.Annotations["cluster.x-k8s.io/restartedAt"]
+					return found
+				}
+				return false
+			}, timeout).WithPolling(pollInterval).Should(BeTrue())
+		})
+		It("Should trigger kubeadmControlPlane rollout when control plane node exists failure domain.", func() {
```

Comment:
I'm not sure I understand this test description. Could you rephrase it? Same with the next test